### ISSUE TITLE
Unsupported URL on theplatform.eu

### ISSUE THREAD

**Issue by runch-randa** [ytdl-org/youtube-dl#24795]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.03.24**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[generic] bcl_4591?form=html: Requesting header
WARNING: Could not send HEAD request to https://player.theplatform.eu/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?form=html: HTTP Error 405: Method Not Allowed
[generic] bcl_4591?form=html: Downloading webpage
WARNING: Falling back on generic information extractor.
[generic] bcl_4591?form=html: Extracting information
[generic] bcl_4591?autoPlay=false: Requesting header
WARNING: Could not send HEAD request to https://player.theplatform.eu/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?autoPlay=false: HTTP Error 405: Method Not Allowed
[generic] bcl_4591?autoPlay=false: Downloading webpage
WARNING: Falling back on generic information extractor.
[generic] bcl_4591?autoPlay=false: Extracting information
ERROR: Unsupported URL: https://player.theplatform.eu/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?autoPlay=false
```

## Description

I'm having issues trying to download from theplatform.eu, one of the supported websites on youtube-dl. [Another user has brought up the same issue already almost half a year ago](https://github.com/ytdl-org/youtube-dl/issues/22798), but to no response. The video plays just fine on the browser, but I get the error above when I try to download using youtube-dl.

Example link: https://player.theplatform.eu/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?form=html

**Comment by willbeaufoy** [comment]
The immediate reason is that youtube-dl is set up to use theplatform.com not theplatform.eu, e.g. we get past the initial error if we change .com to .eu [here](https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/extractor/theplatform.py#L129). However we then run into another error - I played around with it a bit but each time one error is fixed I run into another one.

I tried downloading the link you provided but replacing .eu with .com in the URL, but as I'm in Europe it wouldn't let me. I also tried with my VPN set to the US but for some reason this wouldn't work either ('Temporary failure in name resolution', which I don't think is to do with youtube-dl). So if anyone in the US/with a working VPN can try this command I'd be interested to know if it works:

`youtube-dl -v --simulate https://player.theplatform.com/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?autoPlay=false`

**Comment by BrendanMcCallum** [comment]
@willbeaufoy I don't believe theplatform.com and theplatform.eu have hosting parity (or this specific video is only hosted on the .eu site). I am in the US, but testing like you suggested did not work for me.

I was able to download the link @runch-randa was using, but only up to format http-1901; everything else error-ed out, and I'm not sure why. I _was_ able to download in the highest quality by opening the link in my browser, retrieving the link to the video's master m3u8 playlist from the network monitor, and giving youtube-dl that link. This requires further testing, although my current thought is streams above a certain bit rate can't be downloaded directly and must be streamed.  Possibly a low-level DRM? I will look into if I can fallback to downloading from the m3u8.

I changed all of the regex on the url to match .com **and** .eu.  It seems to work, although this is my first time contributing to the project and I'm not sure if this is the best way to do things even though my changes passed flake8.

A number of other extractors depend on classes in theplatform extractor, and I'm not comfortable making a pull request until I've properly tested that my changes don't break those extractors. That being said, I have forked the repo and made a branch for theplatform. [Here is a direct link to the modified 'theplatform' extractor](https://github.com/BrendanMcCallum/youtube-dl/blob/theplatform/youtube_dl/extractor/theplatform.py).  Please let me know if it works for you.

**Comment by willbeaufoy** [comment]
Hey Brendan, I cloned your repo and checked out the theplatform branch. When I run the following command:

`python3 -m youtube_dl -v https://player.theplatform.eu/p/lCpzgC/dt-vp/embed/select/media/bcl_4591?autoPlay=false`

It starts ok but then I get an error 'Did not get any data blocks'.

Looks like you've solved the URL issue at least though.

**Comment by 89z** [comment]
Issue should be closed, URL is dead
